### ISSUE TITLE
Hellfire Forge most recipes broken

### ISSUE THREAD

**Issue by 3jlowes** [Darkosto/SevTech-Ages#1341]
## Current Behavior

A lot of recipes for blood magic are broken. By broken I mean that some straight up dont work and others work, but they have multiple recipes (I assume 1 being vanilla blood magic and 1 being added to the pack). I will provide examples of both

https://gyazo.com/287bdf9c077c5cb9898459d50b3ac07f
In that screenshot, crafting the Air Reagent has 2 recipes. A more challenging one, from the pack, and the vanilla blood magic recipe. Obviously, people would choose the easier one that doesn't require elytra and raven feathers.

https://gyazo.com/04ac65dfcb554ad93973ec722de7bb0a
In this GIF I showcase the master routing node having 2 recipes first being vanilla blood magic and second being added to the pack. the added recipe does not work despite having the correct amount of will.
## Possible Solution

Investigate this bug as some custom recipes work and some don't. Also, some blood magic vanilla recipes maybe shouldn't be there.

I love blood magic, and I feel in stage 3 it has a lot of potential to have the best tools/armor at that stage.

## Your Environment

* Modpack Version: 3.0.4
* Java Version: Java 8 161 64 bit
* Launcher Used: Twitch
* Resourcepack Enabled?: no
* Optifine Installed?: yes

**Comment by querns** [comment]
I'll add the Node Router to the list of recipes that don't work in the Hellfire Forge. There appears to be a missing ingredient in the top-left corner:

[https://i.imgur.com/2j8NUcd.png](https://i.imgur.com/2j8NUcd.png)

**Comment by gekkobear** [comment]
Scanning the crafttweaker log, it reads that the top left should be a stick.

"[AVAILABLE][CLIENT][INFO] Adding TartaricForge recipe for: 1xitem.bloodmagic.nodeRouter@0 from: [stick,1xitem.bloodmagic.slate@1,1xitem.dyePowder@4,1xitem.blazeRod@0] minSouls: 400.0, soulDrain: 5.0"

Should that be either minecraft:stick or ore:stickWood and not just stick?

**Comment by gekkobear** [comment]
I did some testing, possible fix.
In scripts/mod_integrations/blood_magic.zs

Node router:
Current:
TartaricForge.addRecipe(
	<bloodmagic:node_router>,
	[<ore:stick>, <bloodmagic:slate:1>, <minecraft:dye:4>, <minecraft:blaze_rod>],
	400, 5
);

Proposed:
TartaricForge.addRecipe(
	<bloodmagic:node_router>,
	[<minecraft:stick>, <bloodmagic:slate:1>, <minecraft:dye:4>, <minecraft:blaze_rod>],
	400, 5
);

Worked in my singleplayer test to make a node router in 3.0.6 with this change.

Master Routing Node (copy change for other recipes removing null:
TartaricForge.addRecipe(
	<bloodmagic:master_routing_node>,
	[metals.iron.block.firstItem, metals.platinum.ingot.firstItem, <bloodmagic:slate:2>, null],
	400, 200
);

Proposed change
TartaricForge.addRecipe(
	<bloodmagic:master_routing_node>,
	[metals.iron.block.firstItem, metals.platinum.ingot.firstItem, <bloodmagic:slate:2>],
	400, 200
);
(removed comma and null)

This also worked, I was able to use the updated recipe.
Removing ", null" from the remove recipes removed the standard Air Sigil and Master Routing Node correctly.

The removal of the frame parts required an extra change:
TartaricForge.removeRecipe([<minecraft:glass>, <bloodmagic:slate>, <minecraft:dye:4>, null]);
This is before the updated frame parts recipe with a silver ingot, I think this should be:
TartaricForge.removeRecipe([<minecraft:glass>,, <minecraft:stone>,<bloodmagic:slate>]);

The recipe here was incorrect for the standard frame parts (dye:4 or lapis is in the new recipe, not the old one).
That change removes the standard Frame Parts, leaving only the new/updated one to be crafted.
And testing shows that does craft correctly (it was already correct for the new recipe, as it had 4 pieces and no null).
